asm-utils/src/data_parser.rs

```diff
@@ -14,76 +16,134 @@ pub enum DataValue {
 
 impl DataValue {
     /// Returns the size of the value in bytes.
-    pub fn size(&self) -> usize {
+    ///
+    /// The address is necessary because the size of the alignment padding
+    /// depending on what address it is defined on.
+    pub fn size(&self, from_addr: usize) -> usize {
         match self {
             DataValue::Direct(data) => data.len(),
             DataValue::Zero(length) => *length,
+            DataValue::Alignment(bytes) => alignment_size(from_addr, *bytes),
             DataValue::Reference(_) => 4,
             DataValue::Offset(..) => 4,
         }
     }
 }
 
+#[derive(Default)]
+struct DataSections {
+    sections: Vec<Vec<(Option<String>, Vec<DataValue>)>>,
```

Comment:
Can you add a comment explaining what this represents? Is it a "sequence of sectors, where a sector is a maybe named sequence of values"?